doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,149 @@
+\snippet dna2_only_rank.cpp alphabet_concept
+
+\note
+A `static_assert()` will only tell you whether the expression is true or not.
+If you want the compiler to tell you **why** the concept is not modelled,
+you can construct a dummy requirement like this:
+\snippet dna2_alphabet.cpp dummy_requirement
+This will fail with a (slightly long and need-to-get-used-to) error message telling you
+that `foo()` cannot be called with `dna2` because `constraints are not satisfied ... `.
+
+## Prerequisites
+
+A look at the documentation of seqan3::Alphabet will reveal that it is actually a refinement of other concepts,
+more precisely seqan3::Semialphabet which in turn refines std::CopyConstructible and std::StrictTotallyOrdered.
+Let's check those:
+\snippet dna2_only_rank.cpp other_concepts
+
+You should see that your type models only the std::CopyConstructible concept.
+Let's have a look at the documentation for std::StrictTotallyOrdered. Can you guess what it describes?
+
+It describes the requirements for types that are comparable via `<`, `<=`, `>` and `>=`.
+This is useful so that you can sort a text over the alphabet, for example.
+Additionally, std::StrictTotallyOrdered is again a refinement of std::EqualityComparable,
+which requires the `==` and `!=` operators, so let's first implement those with free member functions.
+\snippet dna2_semialphabet.cpp equality
+
+\assignment{Excercise}
+Implement the inequality operator (!=) for `dna2`.
+\endassignment
+\solution
+\snippet dna2_semialphabet.cpp inequality
+\endsolution
+
+We see that our type now models std::EqualityComparable, which is a prerequisite of std::StrictTotallyOrdered.
+\snippet dna2_comparison_operators.cpp equality_comparable_concept
+
+\assignment{Excercise}
+Implement the four comparison operators *as free member functions* and verify that your type models
+std::StrictTotallyOrdered.
+\endassignment
+\solution
+\snippet dna2_comparison_operators.cpp comparison
+\endsolution
+
+## Semialphabet
+
+Let's move on to the more interesting concepts. seqan3::Semialphabet requires the *rank interface*
+that we have introduced in the [alphabet tutorial](\ref tutorial_alphabets). If we define the member functions
+`to_rank`, `assign_rank` and `alphabet_size` for our type, SeqAn automatically exposes them to the respective global
+functions `seqan3::to_rank`, `seqan3::assign_rank_to` and `seqan3::alphabet_size`.
+Note that `assign_rank` is only required for seqan3::WritableSemialphabet.
+\snippet dna2_semialphabet.cpp semialphabet
+
+As you can see from the `static_assert` our dna2 alphabet now models seqan3::Semialphabet and
+seqan3::WritableSemialphabet:
+\snippet dna2_semialphabet.cpp writable_semialphabet_concept
+
+With this we can automatically use some free functions on our dna2 alphabet. If you are interested in how this works,
+read our section on [Customisation points](\ref about_customisation). Here is an example:
+\snippet dna2_semialphabet.cpp free_functions
```

Comment:
I see that you cover it here, but I think it's better to argue the other way around:
Because the concept requires seqan3::to_char, we have to implement a to_char() member (or other implementation).